src/worker.js

```diff
@@ -54,16 +39,19 @@ create().onRequest('job', ({ contents, type, config, filePath, projectPath, rule
   const configPath = Helpers.getConfigPath(fileDir)
   const relativeFilePath = Helpers.getRelativePath(fileDir, filePath, config)
 
-  const argv = Helpers.getArgv(type, config, rules, relativeFilePath, fileDir, configPath)
+  const cliEngineOptions =
+    Helpers.getCLIEngineOptions(type, config, rules, relativeFilePath, fileDir, configPath)
 
-  if (type === 'lint') {
-    job.response = lintJob(argv, contents, eslint, configPath, config)
+  const noProjectConfig = (configPath === null || isConfigAtHomeRoot(configPath))
+  if (noProjectConfig && config.disableWhenNoEslintConfig) {
+    job.response = []
+  } else if (type === 'lint') {
+    const report = lintJob({ cliEngineOptions, contents, eslint, filePath })
+    job.response = report.results.length ? report.results[0].messages : []
   } else if (type === 'fix') {
-    job.response = fixJob(argv, eslint)
+    job.response = fixJob({ cliEngineOptions, contents, eslint, filePath })
```

Comment:
Actually, it's because I didn't realize `contents` was always `undefined`. 😄 

It's true that ESLint doesn't allow `--fix` from stdin on the command line, but this isn't a problem when calling `executeOnText` because it accepts a [`filename`](https://github.com/eslint/eslint/blob/f560c0660d9ab47c5592b0dd019404ba55f7b6ec/lib/cli-engine.js#L691) argument. So using `executeOnText` followed by `CLIEngine.outputFixes` would allow us to pass unsaved text (if we wanted to do that).

Notably, `executeOnText` also accepts a [warnIgnored](https://github.com/eslint/eslint/blob/f560c0660d9ab47c5592b0dd019404ba55f7b6ec/lib/cli-engine.js#L692) argument, which can be used to disable "file ignored because of matching ignore pattern" messages. If we switch to using `contents` and used `executeOnText` instead of `executeOnFiles`, we could also resolve the issue that you brought up in https://github.com/AtomLinter/linter-eslint/pull/873#discussion_r110074237 without needing to bring back the message list.

(I suppose another way to do this would be to read the file from the filesystem ourselves and call `executeOnText` on it to get rid of the ignore warnings.)